### DISCUSSION TITLE
Where are images stored?

### DISCUSSION THREAD

**Discussion by mblarsen** [imgproxy/imgproxy#1320] [Q&A] [answered]
I run the imgproxy in Azure, but I'm confused for the need for Blob Storage, is this only for reading files to process from there or is this also to let imgproxy us it as storage?

Suppose I process an image from example.com/image.png will it then be stored on the imgproxy server or will it be stored on Blob Storage?

Or is just transiently stored in the server relying on CDN to cache it for as long as it takes?

**Comment by DarthSim** [comment, marked as answer]
Hey @mblarsen,

imgproxy doesn't store the results anywhere. It relies on an external cache (CDN/Nginx/Varnish/etc)

**Reply by mblarsen** [reply]
So every time the original image is used imgproxy will redownload is that correct?

**Reply by DarthSim** [reply]
Yes, that's correct. imgproxy is built to be completely stateless

**Reply by tarun-verma** [reply]
Hey @DarthSim, I'm aware that this is an old issue, but I was wondering if there's any way to instruct imgproxy to store resized images back to the S3 bucket which it's fetching from. So for example, if I have a 1000x1000 image, and I request a 300x300 copy of it, I was wondering if it's possible to store this 300x300 image back to the S3 store as well.

**Reply by DarthSim** [reply]
Hey @tarun-verma!
Currently, it's not possible. We plan to add S3/GCS/etc cache support to the next major release, but it will be meant only for caching purposes, not for direct access to processed images.

**Comment by tacman** [comment]
Looks like version 4 supports this now -- YAY!!
